Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -1076,4 +1076,25 @@ internal struct TextFormatScanner {
         }
         throw TextFormatDecodingError.malformedText
     }
+
+    // This is fast, but only safe for arrays where the
+    // elements never have stray commas or close square brackets.
+    // In particular, never use this for arrays of strings!
+    internal mutating func countArrayElementsQuickly() throws -> Int {
+        let start = p
+        var count = 0
+        while p != end {
+            switch p[0] {
+            case asciiCloseSquareBracket:
+                p = start
+                return count
+            case asciiComma:
+                count += 1
```

Comment:
Are comments a problem here?  A comment might contain extra commas, which would bloat the count here.  The only bad effect would be over-allocating the resulting array.

Fortunately, JSON doesn't allow comments, so this cannot happen there.

Reply:
Actually, does array mode support newlines?  i.e. -
```
foo: [ 1, 2,
  3, 4]
```
If it doesn't, then there is no way for a comment.  If it does, you could use flag to track being in a comment and once it starts, skip until the end of line.

Reply:
Did you get a chance to double check this support?

Reply:
I'm pretty sure comments can occur here, so this does need to be fixed up to avoid overestimating.